### PR TITLE
feat(1click): implemented way to set interest deposit form

### PR DESCRIPTION
**Pull request by milan-bc**
## Description (optional)
This PR brings 1-click interest improvement that we fill up interest deposit form

## Testing Steps (optional)
- Tier 2 account
- go over simple buy and buy some crypto for $300
- once when you click on banner interest form should be pre-fill with data

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.utils.ts [hunk 1]
```diff
@@ -306,10 +307,108 @@ export default ({ coreSagas, networks }: { coreSagas: any; networks: any }) => {
...
+        const xlmAcccount = selectors.components.simpleBuy
+          .getSBBalances(state)
+          .map(x => ({
+            ...x.XLM,
+            coin,
+            address: accountAddress ? accountAddress.data : null
+          }))
+          .map(toCustodialDropdown)
+        defaultAccountR = xlmAcccount.map(head)
+        break
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.utils.ts:399]: all of logic in the switch cases are identical except for the coin map.  i think you make it more generic.  The only real check you need to do is if the coin is BTC or BCH, then you need to also select the `defaultAccountIndex`

**Review by schnogz — CHANGES_REQUESTED** [review]
In general it looks okay but I dont want to add anymore `sb` or `simpleBuy` to the codebase. Lets replace these with `buySell` or something similiar

File: packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts [hunk 2]
```diff
@@ -127,10 +128,15 @@ export default ({
...
   }
 
-  const fetchInterestAccount = function * (coin) {
+  const fetchInterestAccount = function * ({
+    coin
+  }: ReturnType<typeof A.fetchInterestAccount>) {
     try {
       yield put(A.fetchInterestAccountLoading())
-      const paymentAccount = yield call(api.getInterestAccount, coin)
+      const paymentAccount: ReturnType<typeof api.getInterestAccount> = yield call(
```
[the remark this answers is not in the document]
Comment (milan-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts:136]: 👍

File: packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts [hunk 3]
```diff
@@ -261,26 +267,85 @@ export default ({
   }: ReturnType<typeof A.initializeDepositForm>) {
     const { coin, currency } = payload
 
+    const isFromSB = S.getIsFromSB(yield select())
+    if (isFromSB) {
+      // re-fetch the SB balances to have latest since SB flow did changed it
```
[the remark this answers is not in the document]
Comment (milan-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts:272]: ok

File: packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.utils.ts (hunk 1 above)
Comment (milan-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.utils.ts:399]: Yes will improve it ;)

**Review by schnogz — APPROVED** [review]

**Pull request merged**